packages/amplify-sdk/src/auth.js

```diff
@@ -269,35 +290,48 @@ export default class Auth {
 		}
 
 		// copy over the correct auth params
-		for (const prop of [ 'baseUrl', 'clientId', 'realm', 'env' ]) {
+		for (const prop of [ 'baseUrl', 'clientId', 'realm', 'env', 'clientSecret', 'username', 'password', 'secret' ]) {
 			if (account.auth[prop] && opts[prop] !== account.auth[prop]) {
 				log(`Overriding "${prop}" auth param with account's: ${opts[prop]} -> ${account.auth[prop]}`);
 				opts[prop] = account.auth[prop];
 			}
 		}
 		authenticator = this.createAuthenticator(opts);
 
-		if (account.auth.expired) {
-			// refresh the access token if the refresh token is valid
+		const { access, refresh } = account.auth.expires;
+		let doRefresh = account.auth.expired;
+		if (doRefresh) {
 			log(`Access token for account ${highlight(account.name || account.hash)} has expired`);
-
-			const { access, refresh } = account.auth.expires;
 			if ((refresh || access) < Date.now()) {
 				log(`Unable to refresh access token for account ${highlight(account.name || account.hash)} because refresh token is also expired`);
 				return;
 			}
+		} else {
+			const expiresIn = (refresh || access) - Date.now();
+			if (this.tokenRefreshThreshold && (refresh || access) - Date.now() < this.tokenRefreshThreshold * 1000) {
```

Comment:
You should probably re-use your `expiresIn` variable in the if-condition. :)
```javascript
const expiresIn = (refresh || access) - Date.now();
if (this.tokenRefreshThreshold && expiresIn < this.tokenRefreshThreshold * 1000) {
  // ...
}
```

Reply:
Doh! Fixed! Thank you!